Src/ILGPU/IR/IRType.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections.Immutable;
+
+namespace ILGPU.IR
+{
+    /// <summary>
+    /// A uniform value type representing an exported <see cref="Types.TypeNode"/> from an <see cref="IRContext"/>.
+    /// </summary>
+    /// <param name="Id">Corresponds to <see cref="Node.Id"/></param>
+    /// <param name="Class">A <see cref="Classifier"/> value denoting which kind of <see cref="Types.TypeNode"/> this instance represents</param>
+    /// <param name="Nodes">A list of <see cref="NodeId"/> values corresponding to those of related instances; important for later reconstruction</param>
+    /// <param name="BasicValueType">Corresponds to <see cref="Types.TypeNode.BasicValueType"/></param>
+    /// <param name="Data">Extra data specific to this type's kind or instance</param>
+    public record struct IRType(
```

Comment:
In runtime repo people avoid records because records tend to bloat resulting binaries. Not sure if this is a concern here though. Just mentioning.

Reply:
In this case I'm purposefully using `record` so that it auto-implements `IEquatable` and all that.